File: Python/list/Longest Strictly Increasing or Strictly Decreasing Subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        maxl =0
        #Increasing
        for i in range(len(nums)):
            currl=1
            for j in range(i+1,len(nums)):
                if nums[j]>nums[j-1]:
                    currl+=1
                else:
                    break
            maxl=max(maxl,currl)
        
        #Decreasing
        for i in range(len(nums)):
            currl=1
            for j in range(i+1,len(nums)):
                if nums[j]<nums[j-1]:
                    currl+=1
                else:
                    break
            maxl=max(maxl,currl)
        return maxl

File: CPP/Array/Longest Strictly Increasing or Strictly Decreasing Subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int maxl=0;
        //Increasing
        for(int i=0;i<nums.size();i++){
            int currl=1;
            for (int j=i+1;j<nums.size();j++){
                if(nums[j]>nums[j-1]){
                    currl++;
                }else{
                    break;
                }
            }
            maxl=max(maxl,currl);
        }
        //Decreasing
        for(int i=0;i<nums.size();i++){
            int currl=1;
            for (int j=i+1;j<nums.size();j++){
                if(nums[j]<nums[j-1]){
                    currl++;
                }else{
                    break;
                }
            }
            maxl=max(maxl,currl);
        }
        return maxl;
    }
};
